File: src/ros2_bridge/mappers/nav2_msgs/collision_monitor_state.rs
//! Typed mapper for `nav2_msgs/msg/CollisionMonitorState`.

use crate::ros2_bridge::mapper::TypedTopicMapper;

pub(crate) fn collision_monitor_state_to_bus(msg: ros_env::nav2_msgs::msg::CollisionMonitorState) -> crate::nav2_msgs::msg::v1::CollisionMonitorState {
    crate::nav2_msgs::msg::v1::CollisionMonitorState {
        action_type: msg.action_type,
        polygon_name: crate::ros2_bridge::mappers::convert::from_ros_string(msg.polygon_name),
    }
}

pub(crate) fn collision_monitor_state_to_ros(bus: crate::nav2_msgs::msg::v1::CollisionMonitorState) -> ros_env::nav2_msgs::msg::CollisionMonitorState {
    ros_env::nav2_msgs::msg::CollisionMonitorState {
        action_type: bus.action_type,
        polygon_name: crate::ros2_bridge::mappers::convert::to_ros_string(bus.polygon_name),
    }
}

#[derive(Clone, Copy, Debug, Default)]
pub struct Nav2MsgsCollisionMonitorStateMapper;

impl TypedTopicMapper for Nav2MsgsCollisionMonitorStateMapper {
    type Ros = ros_env::nav2_msgs::msg::CollisionMonitorState;
    type Bus = crate::nav2_msgs::msg::v1::CollisionMonitorState;

    fn type_name(&self) -> &'static str {
        "nav2_msgs/msg/CollisionMonitorState"
    }

    fn ros_to_bus(&self, msg: Self::Ros) -> crate::errors::Result<Self::Bus> {
        Ok(collision_monitor_state_to_bus(msg))
    }

    fn bus_to_ros(&self, msg: Self::Bus) -> crate::errors::Result<Self::Ros> {
        Ok(collision_monitor_state_to_ros(msg))
    }
}

File: bindings/python/robot_bus/ros2_bridge/mappers/nav2_msgs/collision_monitor_state.py
"""Generated mapper for `nav2_msgs/msg/CollisionMonitorState`."""

from __future__ import annotations

from robot_bus.ros2_bridge.mappers import _convert

_BusMsg = None


def _bus_cls():
    global _BusMsg
    if _BusMsg is None:
        from robot_bus.nav2_msgs.msg.v1 import CollisionMonitorState as BusMsg

        _BusMsg = BusMsg
    return _BusMsg


def collision_monitor_state_to_bus(msg):
    BusMsg = _bus_cls()
    bus = BusMsg()
    bus.action_type = msg.action_type
    bus.polygon_name = str(msg.polygon_name)
    return bus


def collision_monitor_state_to_ros(bus):
    from nav2_msgs.msg import CollisionMonitorState as RosMsg

    out = RosMsg()
    out.action_type = bus.action_type
    out.polygon_name = str(bus.polygon_name)
    return out


class Nav2MsgsCollisionMonitorStateMapper:
    _ros_type = None

    def ros_msg_type(self):
        cls = type(self)
        if cls._ros_type is None:
            from nav2_msgs.msg import CollisionMonitorState as RosMsg

            cls._ros_type = RosMsg
        return cls._ros_type

    def ros_to_bus(self, msg) -> bytes:
        return collision_monitor_state_to_bus(msg).SerializeToString()

    def bus_to_ros(self, payload: bytes):
        BusMsg = _bus_cls()
        bus = BusMsg()
        bus.ParseFromString(payload)
        return collision_monitor_state_to_ros(bus)
